packages/graphql/src/index.ts

```diff
@@ -1,24 +1,26 @@
 import { createResource, ResourceReturn } from "solid-js";
 import { access, MaybeAccessor, Modify } from "@solid-primitives/utils";
+import { TypedDocumentNode } from "@graphql-typed-document-node/core";
+import { DocumentNode, print } from "graphql";
 
-export type RequestOptions = Modify<
+export type RequestOptions<V = unknown> = Modify<
   Omit<RequestInit, "body">,
   {
     headers?: RequestHeaders;
-    variables?: QueryVariables;
+    variables?: V;
     fetcher?: typeof fetch;
   }
 >;
 
 export type GraphQLClientQuery = {
-  <T = unknown>(
-    query: string,
-    variables: MaybeAccessor<QueryVariables | boolean> | undefined,
```

Comment:
I know that's not your doing, but allowing for `boolean` means that `true` could also be passed to bypass passing variables. So a `FalsyValue` from [utils](https://github.com/solidjs-community/soliprimitives/blob/main/packages/utils/src/types.ts#L116) should be a better fit.

Reply:
I'll check it